### PR TITLE
PT-1.12 CVE patching

### PR DESCRIPTION
**Pull request by ShiboXing**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Tests run

**NOTE: By default, docker builds are disabled. In order to build your container, please update dlc_developer_config.toml and specify the framework to build in "build_frameworks"**
- [ ] I have run builds/tests on commit <INSERT COMMIT ID> for my changes.

**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by ShiboXing** [comment]
standard: f23d62a

File: pytorch/training/docker/1.12/py3/Dockerfile.cpu [hunk 1]
```diff
@@ -208,6 +208,9 @@ RUN HOME_DIR=/root \
  && rm -rf ${HOME_DIR}/oss_compliance* \
  && rm -rf /tmp/tmp*
 
+# Removing the cache as it is needed for security verification
+RUN rm -rf /root/.cache | true
```
Comment (kace) [inline pytorch/training/docker/1.12/py3/Dockerfile.cpu:212]: Not sure if I understand why we need `| true`. Can you explain?

File: pytorch/training/docker/1.12/py3/cu113/Dockerfile.gpu [hunk 2]
```diff
@@ -180,7 +180,7 @@ RUN ompi_info --parsable --all | grep mpi_built_with_cuda_support:value \
     conda-content-trust \
     charset-normalizer \
     "shap<1" \
- && /opt/conda/bin/conda install -c dglteam -y dgl-cuda11.3 \
+ && /opt/conda/bin/conda install -c dglteam -y dgl-cuda11.3=0.9.1 \
```
Comment (kace) [inline pytorch/training/docker/1.12/py3/cu113/Dockerfile.gpu:183]: Why are we specifying version?

File: pytorch/training/docker/1.12/py3/cu113/Dockerfile.gpu [hunk 3]
```diff
@@ -456,9 +459,10 @@ RUN mkdir -p /etc/pki/tls/certs && cp /etc/ssl/certs/ca-certificates.crt /etc/pk
 WORKDIR /
 
 # Install scikit-learn and pandas
-RUN conda install -y -c conda-forge \
+RUN /opt/conda/bin/conda install -y -c conda-forge \
     scikit-learn \
-    pandas
+    pandas \
+ && /opt/conda/bin/conda clean -ya
```
Comment (kace) [inline pytorch/training/docker/1.12/py3/cu113/Dockerfile.gpu:465]: What does `conda clean` do? Why do we need it?

Comment (ShiboXing) [reply pytorch/training/docker/1.12/py3/cu113/Dockerfile.gpu]: initially thought to remove the root cache, but just removed it!

File: pytorch/training/docker/1.12/py3/cu113/Dockerfile.gpu (hunk 2 above)
Comment (ShiboXing) [reply pytorch/training/docker/1.12/py3/cu113/Dockerfile.gpu]: They had a recent release of 0.9.1post1, but the upstream test has caught up and failed. 
This morning when I tried the test it seemed to work fine. Trying a build with dgl unpinned.

**Review by kace — APPROVED** [review]

**Review by kace — APPROVED** [review]

**Pull request merged**